pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/Connection.java

```diff
@@ -105,6 +108,29 @@ public ResultSetGroup execute(Request request)
     return execute(null, request);
   }
 
+  /**
+   * Extract tableName from the given query
+   * @param query
+   * @return tableName
+   * @throws PinotClientException 
+   */
+  public String getTableNameFromQuery(String query) {
+    PinotQuery pinotQuery;
+
+    try {
+      pinotQuery = CalciteSqlParser.compileToPinotQuery(query);
+    } catch (PinotClientException e) {
+      LOGGER.error("Caught exception while compiling SQL request: {}, {}", query, e.getMessage());
```

Comment:
We don't want to log an error here because this is most likely a bad query. Suggest not logging because user can get the `PinotClientException`